src/elm/Profile/Contact.elm

```diff
@@ -0,0 +1,202 @@
+module Profile.Contact exposing
+    ( Contact
+    , Country(..)
+    , Normalized
+    , countryToFlag
+    , decode
+    , encode
+    , hasSameType
+    , listCountries
+    , normalize
+    , selectionSet
+    , unwrap
+    , usesPhone
+    , validator
+    )
+
+import Cambiatus.Enum.ContactType as ContactType exposing (ContactType)
+import Cambiatus.Object
+import Cambiatus.Object.Contact
+import Graphql.SelectionSet as SelectionSet exposing (SelectionSet, with)
+import Json.Decode as Decode exposing (Decoder)
+import Json.Encode as Encode
+import Regex exposing (Regex)
+import Session.Shared exposing (Translators)
+import Validate
+
+
+
+-- MODEL
+
+
+type alias Contact =
+    { contactType : ContactType
+    , contact : String
+    }
+
+
+{-| The contact string must be formatted in a specific way to be sent to the
+backend, so we have this type to ensure the information is normalized
+-}
+type Normalized
+    = Normalized Contact
```

Comment:
This is so we can be sure we send valid data to the server, such as adding `https://t.me/`, or a country code before the contact string. I guess we could `normalize` the data just before sending it to the server, in `Profile.mutation`, but then we'd possibly need to carry a `Country` all the way through. Also, if we need other functions to manipulate a contact, it might be useful to know if it's been normalized or not. This is sort of like the [Wrap early, unwrap late](https://sporto.github.io/elm-patterns/basic/wrap-early.html) and [Parse don't validate](https://sporto.github.io/elm-patterns/basic/parse-dont-validate.html) patterns.

Reply:
but in this module here we don't do such a thing right? The Normalized type is never used to format data, we only unwrap it, but never with a specific unwrapping rule (like the example where they put a `"US$"`prefix. 

Premature optimization is fine, but on Elm its best to add it as we need. Great talk about this subject: https://www.youtube.com/watch?v=DoA4Txr4GUs

Reply:
It is used to format data, but the data is formatted on wrap, instead of unwrap (maybe formatting on unwrap is a better idea?). It's like a tag that says "Hey, this data has been formatted, and it can be trusted", just like `Valid` from [`rtfeldman/elm-validate`](https://package.elm-lang.org/packages/rtfeldman/elm-validate/latest/). It stops us, developers, from sending something like "myprofile" instead of "https://instagram.com/myprofile" to the backend, because `ProfileForm`, (and by extension `Profile.mutation`) expects a `Normalized` contact. I still think that would be a good use case, but if you disagree, just let me know!

Reply:
nah, ok, keep it that way 🙂